pkg/mdm/mdmtest/windows.go

```diff
@@ -0,0 +1,468 @@
+	// Get SessionID
+	sessionID, err := c.lastManagementResp.GetSessionID()
+	if err != nil {
+		return nil, fmt.Errorf("session ID processing error %w", err)
+	}
+
+	// Get MessageID
+	messageID, err := c.lastManagementResp.GetMessageID()
+	if err != nil {
+		return nil, fmt.Errorf("message ID processing error %w", err)
+	}
+	messageIDInt, err := strconv.Atoi(messageID)
+	if err != nil {
+		return nil, fmt.Errorf("converting message ID to int: %w", err)
+	}
+
+	var msg fleet.SyncML
+	msg.Xmlns = microsoft_mdm.SyncCmdNamespace
+	msg.SyncHdr = fleet.SyncHdr{
+		VerDTD:    microsoft_mdm.SyncMLSupportedVersion,
+		VerProto:  microsoft_mdm.SyncMLVerProto,
+		SessionID: sessionID,
+		MsgID:     fmt.Sprint(messageIDInt + 1),
+		Source:    &fleet.LocURI{LocURI: &c.DeviceID},
+		Target: &fleet.LocURI{
+			LocURI: ptr.String(c.fleetServerURL + microsoft_mdm.MDE2ManagementPath),
+		},
+	}
+
+	// iterate over mocked responses and append them to the SyncML message
+	for _, protoCmd := range c.queuedCommandResponses {
+		msg.AppendCommand(fleet.MDMRaw, protoCmd)
+	}
+
+	xmlReq, err := xml.MarshalIndent(msg, "", "\t")
+	if err != nil {
+		return nil, fmt.Errorf("serializing XML req: %w", err)
+	}
+
+	return c.doManagementReq(xmlReq)
+}
+
+// SetResponse sets a response for a specific command UUID.
+func (c *TestWindowsMDMClient) AppendResponse(op fleet.SyncMLCmd) {
+	c.queuedCommandResponses[op.CmdID] = op
+}
+
+func (c *TestWindowsMDMClient) GetCurrentMsgID() (string, error) {
+	msgID, err := c.lastManagementResp.GetMessageID()
+	if err != nil {
+		return "", fmt.Errorf("getting management response msg id: %w", err)
+	}
+	return msgID, nil
+}
+
+func (c *TestWindowsMDMClient) Enroll() error {
+	if err := c.Discovery(); err != nil {
+		return err
+	}
+
+	if err := c.Policy(); err != nil {
+		return err
+	}
+
+	binarySecToken, err := fleet.GetEncodedBinarySecurityToken(c.enrollmentType, c.orbitNodeKey)
+	if err != nil {
+		return fmt.Errorf("generating encoded security token: %w", err)
+	}
+
+	enrollReq := []byte(`
+<s:Envelope
+    xmlns:s="http://www.w3.org/2003/05/soap-envelope"
+    xmlns:a="http://www.w3.org/2005/08/addressing"
+    xmlns:u="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd"
+    xmlns:wsse="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd"
+    xmlns:wst="http://docs.oasis-open.org/ws-sx/ws-trust/200512"
+    xmlns:ac="http://schemas.xmlsoap.org/ws/2006/12/authorization">
+    <s:Header>
+        <a:Action s:mustUnderstand="1">http://schemas.microsoft.com/windows/pki/2009/01/enrollment/RST/wstep</a:Action>
+        <a:MessageID>urn:uuid:0d5a1441-5891-453b-becf-a2e5f6ea3749</a:MessageID>
+        <a:ReplyTo>
+            <a:Address>http://www.w3.org/2005/08/addressing/anonymous</a:Address>
+        </a:ReplyTo>
+        <a:To s:mustUnderstand="1">https://roperzh-fleet.ngrok.io/api/mdm/microsoft/enroll</a:To>
```

Comment:
Just double-checking that this was not left there by mistake (the `roperzh-fleet...` address)? I don't think it matters given that all protocol communication is mocked in this case?

Reply:
wow, great eye! agreed that it doesn't seem to matter, but I updated it.